src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/accessing/SqueakObjectNewNode.java

```diff
@@ -155,10 +157,10 @@ protected static final WeakVariablePointersObject doWeakPointersUncached(final S
         return new WeakVariablePointersObject(image, classObject, null, extraSize);
     }
 
-    @SuppressWarnings("unused")
     @Specialization(guards = "classObject.isEphemeronClassType()")
-    protected static final WeakVariablePointersObject doEphemerons(final SqueakImageContext image, final ClassObject classObject, final int extraSize) {
-        throw SqueakException.create("Ephemerons not (yet) supported");
+    protected static final EphemeronObject doEphemeron(final SqueakImageContext image, final ClassObject classObject, final int extraSize) {
+        assert extraSize == 0;
```

Comment:
If this assertion is correct, `EphemeronObject` can only have a `key` and `value`.